lightningd/plugin.c

```diff
@@ -229,12 +229,11 @@ plugin_request_new_(struct plugin *plugin,
 	    (arg))
 
 /**
- * Given a request, send it to the plugin.
+ * Send a JSON-RPC message (request or notification) to the plugin.
  */
-static void plugin_request_queue(struct plugin *plugin,
-				 struct plugin_request *req)
+static void plugin_send(struct plugin *plugin, struct json_stream *stream TAKES)
 {
-	*tal_arr_expand(&plugin->js_arr) = req->stream;
+	*tal_arr_expand(&plugin->js_arr) = tal_steal(plugin->js_arr, stream);
```

Comment:
Make this two lines: we got bitten by this evaluation order indeterminism before.